ql/src/java/org/apache/hadoop/hive/ql/exec/Registry.java

```diff
@@ -553,21 +553,19 @@ private void addFunction(String functionName, FunctionInfo function) {
         Integer refCount = persistent.get(functionClass);
         persistent.put(functionClass, Integer.valueOf(refCount == null ? 1 : refCount + 1));
       }
+    } catch (ClassNotFoundException e) {
```

Comment:
The issues' root should be removePersistentFunctionUnderLock when trying to get per UDFClass below --  not sure why this is needed here as part of addFunction?

Reply:
getPermanentUdfClass may throw ClassNotFoundException

Reply:
I see, but this is not part of the Drop cascade issue right?

Reply:
got it, just realized its been called from here